sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_ml_client.py

```diff
@@ -606,3 +606,101 @@ def test_enable_telemetry(self) -> None:
+                {},
+                "online_deployments",
+                "begin_create_or_update",
+            ),
+            (
+                [load_batch_endpoint("tests/test_configs/endpoints/batch/batch_endpoint_mlflow.yaml")],
+                {},
+                "batch_endpoints",
+                "begin_create_or_update",
+            ),
+            (
+                [load_batch_deployment("tests/test_configs/deployments/batch/batch_deployment_1.yaml")],
+                {},
+                "batch_deployments",
+                "begin_create_or_update",
+            ),
+        ],
+    )
+    def test_ml_client_begin_create_without_ws_or_registry_name(
+        self, args, kwargs, ops_name, create_method_name, mock_credential
+    ) -> None:
+        ml_client = MLClient(
+            credential=mock_credential,
+            subscription_id=Test_Subscription,
+            resource_group_name=Test_Resource_Group,
+            registry_name=None,
+        )
+
+        with pytest.raises((ValidationException, Exception)) as val_exception:
+            ml_client.begin_create_or_update(*args, **kwargs)
+            ml_client.__getattribute__(ops_name).__getattr__(create_method_name).assert_called_once_with(
+                *args, **kwargs
+            )
+        assert "This operation requires that you specify a workspace" in str(val_exception.value)
+
+    @pytest.mark.parametrize(
+        "args, kwargs, ops_name, create_method_name",
+        [
+            (
+                [load_job("tests/test_configs/pipeline_jobs/helloworld_pipeline_job.yml")],
+                {},
+                "jobs",
+                "create_or_update",
+            ),
+            (
+                [load_model("tests/test_configs/model/model_full.yml")],
+                {},
+                "models",
+                "create_or_update",
+            ),
+            (
+                [load_datastore("tests/test_configs/datastore/blob_store.yml")],
+                {},
+                "datastores",
+                "create_or_update",
+            ),
+            (
+                [load_environment("tests/test_configs/environment/environment_conda.yml")],
+                {},
+                "environments",
+                "create_or_update",
+            ),
+        ],
+    )
+    def test_ml_client_create_update_without_ws_or_registry_name(
+        self, args, kwargs, ops_name, create_method_name, mock_credential
+    ) -> None:
+        ml_client = MLClient(
+            credential=mock_credential,
+            subscription_id=Test_Subscription,
+            resource_group_name=Test_Resource_Group,
+            registry_name=None,
+        )
+
+        with pytest.raises((ValidationException, Exception)) as ex:
+            ml_client.create_or_update(*args, **kwargs)
+            ml_client.__getattribute__(ops_name).__getattr__(create_method_name).assert_called_once_with(
+                *args, **kwargs
+            )
```

Comment:
Same feedback as above.

---

This code will never actually run, since the expectation is that the test throws an exception on the previous line.

I'm assuming that you're not actually interested in testing that the polymorphic create function is routing correctly here, so you could just directly invoke `begin_create_or_update` on the operation class.

```python
            ml_client.__getattribute__(ops_name).__getattr__(create_method_name).assert_called_once_with(
                *args, **kwargs
            )
```

```suggestion
            getattr(ml_client, ops_name).begin_create_or_update(*args, **kwargs)
```